backup-n-restore/hack/common.sh

```diff
@@ -187,3 +187,196 @@ restore_finished() {
     fi
     echo ${rv}
 }
+
+
+# register_managed_clusters modifies the boostrap kubeconfig in every managed cluster. It also creates 
+# creates clusterrole and clusterrolebdings for the <managed cluster name> in th new hub
+#
+register_managed_clusters() {
+    server=$(oc config view -o jsonpath='{.clusters[0].cluster.server}')
+    echo_green "Server -> $server"
+
+    for managedclustername in $(oc get ns -lcluster.open-cluster-management.io/managedCluster -o jsonpath='{.items[*].metadata.name}');
+    do managed_kubeconfig_secret=$(oc get secret -o name -n $managedclustername | grep admin-kubeconfig);
+       if [ -z "${managed_kubeconfig_secret}" ]; then #this will skip local-cluster as well
+	   echo "skipping $managedclustername"
+	   continue
+       fi
+
+       managed_kubeconfig_secret=$(basename $managed_kubeconfig_secret)
+       managedclusterkubeconfig=$(mktemp)
+       oc get secret $managed_kubeconfig_secret -n $managedclustername -o jsonpath={.data.kubeconfig} | base64 -d > $managedclusterkubeconfig
+       
+       server=$(oc config view -o jsonpath='{.clusters[0].cluster.server}')
+       secretname=$(oc get secret -o name -n $managedclustername | grep ${managedclustername}-bootstrap-sa-token)
+       ca=$(kubectl get ${secretname} -n ${managedclustername} -o jsonpath='{.data.ca\.crt}')
+       token=$(kubectl get ${secretname} -n ${managedclustername} -o jsonpath='{.data.token}' | base64 --decode)
+   
+       newbootstraphubkubeconfig=$(mktemp)
+       cat << EOF > ${newbootstraphubkubeconfig}
+apiVersion: v1
+kind: Config
+clusters:
+- name: default-cluster
+  cluster:
+    certificate-authority-data: ${ca}
+    server: ${server}
+contexts:
+- name: default-context
+  context:
+    cluster: default-cluster
+    namespace: default
+    user: default-auth
+current-context: default-context
+users:
+- name: default-auth
+  user:
+    token: ${token}
+EOF
+       
+       echo_yellow "Created $newbootstraphubkubeconfig"
+       
+       oc --kubeconfig=$managedclusterkubeconfig delete secret  bootstrap-hub-kubeconfig -n open-cluster-management-agent
+       
+       oc --kubeconfig=$managedclusterkubeconfig create secret generic bootstrap-hub-kubeconfig --from-file=kubeconfig="${newbootstraphubkubeconfig}" -n open-cluster-management-agent
+
+       rm -rf $managedclusterkubeconfig
+       if [ $? -eq 0 ]
+       then
+           echo_green "tmp kubeconfig $managedclusterkubeconfig deleted"
+       fi
+       
+       rm -rf $newbootstraphubkubeconfig
+       if [ $? -eq 0 ]
+       then
+	   echo_green "tmp kubeconfig $newbootstraphubkubeconfig deleted"
+       fi
+
+       cat << EOF | oc  apply -f -
```

Comment:
@sdminonne You shouldn't have to create the ClusterRoles/ClusterRoleBindings. These are generated by the `ManagedCluster` controller (the `registration` pods).

Reply:
@mdelder I'm not positive about that, but to be honest I lost the logs containing errors that forced me to add these RBACs, I need to retest it.

Reply:
Just a fact, these are clusterrole and clusterrolebindings on the HUB side, since [our](https://github.com/open-cluster-management/sre-tools/blob/main/backup-n-restore/backup-acm.sh#L117) doesn't backup cluster scoped resources we do need to recreate them. Anyway I'll retest all this.